src/ploneintranet/workspace/workspacecontainer.py

```diff
@@ -15,3 +15,12 @@ class WorkspaceContainer(Container):
     A folder to contain WorkspaceFolders
     """
     grok.implements(IWorkspaceContainer)
+
+try:
```

Comment:
Agree with @mattss here. This logic sounds like it should be kept out of pi.workspace if possible.